proposals/lip-state-model.md

```diff
@@ -0,0 +1,105 @@
+```
+LIP: <LIP number>
+Title: State model and state root
+Author: Alessandro Ricottone <alessandro.ricottone@lightcurve.io>
+Type: Informational
+Created: <YYYY-MM-DD>
+Updated: <YYYY-MM-DD>
+Required: 0039
+```
+
+
+## Abstract
+
+The state of an interoperable chain in the Lisk ecosystem is organized as a sparse Merkle tree, the state tree, built on top of a generic key-value map for each module of the chain.
+This allows to authenticate the whole state with the tree Merkle root, the state root.
+In this LIP, we define the state model of a Lisk blockchain and the construction of the state tree from which the state root is calculated.
+
+
+## Copyright
+
+This LIP is licensed under the [Creative Commons Zero 1.0 Universal](https://creativecommons.org/publicdomain/zero/1.0/).
+
+
+## Motivation
+
+A blockchain created with the Lisk SDK is organized in a modular way: each module registered in the chain defines its own state and the possible state transitions, i.e., the transactions defined within the module or the reducers that can be called by other modules.
+This is reflected by the state tree: each module maintains its own key-value map and sparse Merkle tree, which are finally merged together to form the complete state tree.
+This LIP specifies the general format for the key-value maps and the way in which they are combined together in the state tree. 
+
+Organizing the state of a blockchain as a Merkle tree allows to cryptographically authenticate the whole state with a single hash, the state root.
+The state root property is calculated at the end of the block processing and included in the block header. 
+Information from the block header is then used to create a certificate and signed by the chain validators.
+When a certificate is posted to a chain, the state root is used to authenticate the information contained in it.
+For example, a node can verify that the cross-chain messages in the certificate payload have been included in the outbox of the sending chain by recomputing the state root signed in the certificate from the outbox root.
```

Comment:
The cross-chain messages are contained in the CCU transaction, not in the certificate payload. Also, the state root contained a certificate rather authenticates other information in the CCU, not other information in the certificate (previous sentence).

Reply:
rephrased

Reply:
edited